src/actions/actions.ts

```diff
@@ -696,7 +696,9 @@ export class CommandOneNormalCommandInInsertMode extends BaseCommand {
 
   public async exec(position: Position, vimState: VimState): Promise<VimState> {
     vimState.returnToInsertAfterCommand = true;
-    return await new CommandEscInsertMode().exec(position, vimState);
+    return await new CommandEscInsertMode().exec(
```

Comment:
I don't think that this bit is relevant to the D fix, right? Could you remove it and include it in a separate PR?